setup.sh

```diff
@@ -1,36 +1,73 @@
 #!/bin/bash
 
-if [ $(whoami) == root ]; then
-    pkg_manager_cmd="apt-get install -qy" # q provides less verbose o/p. y assumes yes for all prompts.
-else
-    pkg_manager_cmd="sudo apt-get install -qy" # q provides less verbose o/p. y assumes yes for all prompts.
+quiet_mode=false
+
+# Parse command line options
+while [ ! $# -eq 0 ]
+do
+    case "$1" in
+        --help | -h)
+            echo "use --quiet-mode or -q flag to run the script in non-interactive, less verbose mode"
+            exit 0
+            ;;
+        --quiet-mode | -q)
+            quiet_mode=true
+            echo -e "Running setup in quiet mode : non-interactive and less verbose output\n"
+            ;;
+        *)
+            echo "Unknown option provided: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+pkg_manager_cmd="apt-get install"
+
+# if not invoked as root, prefix sudo to package manager command
+if [ ! $(whoami) == root ]; then
+    pkg_manager_cmd="sudo $pkg_manager_cmd"
+fi
+
+# add -qy flags in quiet mode, to provide less verbose output and assume yes for all prompts
+if [ $quiet_mode == true ]; then
+    pkg_manager_cmd="$pkg_manager_cmd -qy"
 fi
 
 check_install_pkg() {
 
-    pkg_name="$1"
+    cmd="$1"
     pkg_version_check_cmd="$2"
-    pkg_install_cmd="$3"
+    pkg_to_install="$3"
 
     if $pkg_version_check_cmd 2>/dev/null; then
-        echo -e "$($pkg_version_check_cmd)pi already installed.\n"
+        echo -e "$($pkg_version_check_cmd) already installed.\n"
```

Comment:
I guess it's only partially fixed. The tree characters `$(` and `)` can still be deleted, can't they?
At any rate, shouldn't the message just output the pkg, which is in variable `$cmd` now:
```suggestion
        echo -e "$cmd already installed.\n"
```

Reply:
$cmd will print only the name of the software. But, this will print the software and the version . It can be useful in cases where setup log is used for analyzing any issue.

Reply:
Ah ok, now I see..